django/test/testcases.py

```diff
@@ -303,7 +303,7 @@ def assertRedirects(self, response, expected_url, status_code=302,
                 path = urljoin(response.request['PATH_INFO'], path)
 
             if fetch_redirect_response:
-                if netloc:
+                if netloc not in ('', 'testserver'):
```

Comment:
@charettes I believe there is no guarantee that those will be the same thing. In this case, 'testserver' is the actual string passed into `client.get()`